src/VisualStudio/Core/Def/Implementation/UnusedReferences/ProjectAssets/ProjectAssetsReader.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.UnusedReferences;
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.UnusedReferences.ProjectAssets
+{
+    // This class will read the dependency heirarchy from the project.assets.json file. The format of this file
+    // is subject to change and in the future this information will be provided by an API.  See https://github.com/dotnet/roslyn/issues/50054
+    internal static class ProjectAssetsReader
+    {
+        // NuGet will include entries to keep empty folders from being removed. These entries can be ignored.
+        private const string NuGetEmptyFileName = "_._";
+
+        public static ImmutableArray<ReferenceInfo> ReadReferences(
+            ImmutableArray<ReferenceInfo> projectReferences,
+            string projectAssetsFilePath,
+            string targetFrameworkMoniker)
+        {
+            if (!File.Exists(projectAssetsFilePath))
+            {
+                return ImmutableArray<ReferenceInfo>.Empty;
+            }
+
+            var projectAssetsFileContents = File.ReadAllText(projectAssetsFilePath);
+            ProjectAssetsFile projectAssets;
+
+            try
+            {
+                projectAssets = JsonConvert.DeserializeObject<ProjectAssetsFile>(projectAssetsFileContents);
+            }
+            catch
+            {
+                return ImmutableArray<ReferenceInfo>.Empty;
+            }
+
+            if (projectAssets is null ||
+                projectAssets.Version != 3)
```

Comment:
They have plans to change the format but not a definite timeline. One of the follow up items is request and move to a dedicated api, since the information we need is already populated in the Solution Explorer.